pyccel/parser/semantic.py

```
@@ -2699,6 +2699,12 @@ def _handle_is_operator(self, IsClass, expr, **settings):
         var1 = self._visit(expr.lhs)
         var2 = self._visit(expr.rhs)
 
+        if (var1 == var2):
```

This should only compare Variable instances. I think this will return `True` for the following case when it should raise an error:
```
In [7]: a+b is a+b
Out[7]: False
```

---

I have just run the test and this is indeed translated to `True`

---

yes you are right

---

@EmilyBourne Good catch!